flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/source/base/StreamResultMatcher.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.pulsar.source.base;
+
+import org.apache.flink.streaming.api.datastream.DataStream;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+
+/** Test results matcher for comparing. */
+public final class StreamResultMatcher<T> extends TypeSafeDiagnosingMatcher<DataStream<T>> {
```

Comment:
I'm not sure if this class isn't overengineered.

I could replace 
```
        List<Integer> expectedResults =
                FOO_LIST.stream().map(SampleData.Foo::getI).collect(toList());

        assertThat(stream, equalsInAnyOrder(expectedResults));
 ```

with

```
        Integer[] expectedResults =
                FOO_LIST.stream().map(SampleData.Foo::getI).toArray(Integer[]::new);

        assertThat(
                stream.executeAndCollect(expectedResults.length),
                Matchers.containsInAnyOrder(expectedResults));
```
and receive the same result with the standard matchers.